### PR TITLE
Json tracing updates and fixes (enable in all clusters, fix lists and large payloads)

### PR DESCRIPTION
**Pull request by andy31415**
- Enables full tracing for all-clusters-app as it is a very common test app
- Increases size for hex buffer to 4096 as we have found messages that are very large to potentially need encoding
- Fix duplicate key bug in json tracing (all keys in a list are "[]" and this meant at least for lists, json tracing was losing data)

### CODE REVIEW

File: src/tracing/json/json_tracing.cpp [hunk 1]
```diff
@@ -63,12 +63,25 @@ ::Json::Value GetPayload(PayloadDecoderType & decoder)
...
     StringBuilder<128> formatter;
 
+    int unique_idx = 0;
+
     while (decoder.Next(entry))
     {
         switch (entry.GetType())
         {
         case PayloadEntry::IMPayloadType::kNestingEnter:
             formatter.Reset().Add(entry.GetName()); // name gets destroyed by decoding
```
Comment (bzbarsky-apple) [inline src/tracing/json/json_tracing.cpp:73]: This "name gets destroyed by decoding" comment does not mesh well for me with the later GetName() calls.  What is this comment saying, exactly?

File: src/tracing/json/json_tracing.cpp [hunk 2]
```diff
@@ -63,12 +63,25 @@ ::Json::Value GetPayload(PayloadDecoderType & decoder)
...
+
+            while (value.isMember(formatter.c_str()))
```
Comment (bzbarsky-apple) [inline src/tracing/json/json_tracing.cpp:75]: This code is not very clear....  Please document that this is trying to find a name that is not already being used as a key in "value"?

File: src/tracing/json/json_tracing.cpp [hunk 3]
```diff
@@ -84,7 +97,17 @@ ::Json::Value GetPayload(PayloadDecoderType & decoder)
             value[formatter.Reset().AddFormat("EVNT(%u/%u)", entry.GetClusterId(), entry.GetEventId()).c_str()] = "<NOT_DECODED>";
             continue;
         default:
-            value[entry.GetName()] = entry.GetValueText();
+            formatter.Reset().Add(entry.GetName()); // name gets destroyed by decoding
```
Comment (bzbarsky-apple) [inline src/tracing/json/json_tracing.cpp:100]: Again, please document.  And what does the "destroyed" comment mean?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/tracing/json/json_tracing.cpp (hunk 1 above)
Comment (andy31415) [reply src/tracing/json/json_tracing.cpp]: Updated code a bit here.

The issue I have is that the current name is from the "decode tree" and as we traverse the decode tree inGetPayload, the current name may change. Not sure how to express that.

File: src/tracing/json/json_tracing.cpp (hunk 3 above)
Comment (andy31415) [reply src/tracing/json/json_tracing.cpp]: This does not apply here. Comment removed.

File: src/tracing/json/json_tracing.cpp (hunk 1 above)
Comment (andy31415) [reply src/tracing/json/json_tracing.cpp]: Updated comment. Hoping this is clearer. Issue is that any returned entries share a common "name buffer" to format values, so if the value requires formatting (not always as names are const c_str, but may happen for dynamic names) a call of GetPayload clobbers the previous name value.

File: src/tracing/json/json_tracing.cpp [hunk 4]
```diff
@@ -56,33 +56,89 @@ using namespace chip::Decoders;
 
 using PayloadDecoderType = chip::Decoders::PayloadDecoder<64, 256>;
 
+/// Figures out a unique name within a json object.
+///
+/// Decoded keys may be duplicated, like list elements are denoted as "[]".
+/// The existing code does not attempt to encode lists and everytying is an object,
```
Comment (bzbarsky-apple) [inline src/tracing/json/json_tracing.cpp:62]:
```suggestion
/// The existing code does not attempt to encode lists and everything is an object,
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/tracing/json/json_tracing.cpp [hunk 5]
```diff
@@ -56,20 +56,89 @@ using namespace chip::Decoders;
...
+
+    void NextName(const char * baseName)
+    {
+        if (strcmp(baseName, "[]") == 0)
+        {
+            mFormatter.Reset().Add("[").Add(mUniqueIndex++).Add("]");
+        }
+        else
+        {
+            mFormatter.Reset().Add(baseName).Add(" - ").Add(mUniqueIndex++);
```
Comment (tcarmelveilleux) [inline src/tracing/json/json_tracing.cpp:111]: Suggest `@unique_id` rather than `- unique_id`

File: src/tracing/json/json_tracing.cpp [hunk 6]
```diff
@@ -56,20 +56,89 @@ using namespace chip::Decoders;
...
+                ChipLogError(Automation, "Potential data loss: insufficient space for unique keys in json");
+                return;
+            }
+        }
+    }
+
+private:
+    void FirstName(const char * baseName)
+    {
+        if (strcmp(baseName, "[]") == 0)
```
Comment (tcarmelveilleux) [inline src/tracing/json/json_tracing.cpp:93]: Suggest replacing anonymous tag with something that doesn't look like list indexing.

File: src/lib/format/protocol_decoder.cpp [hunk 7]
```diff
@@ -292,7 +292,15 @@ void PayloadDecoderBase::ExitContainer(PayloadEntry & entry)
         {
             mPayloadPosition.Exit();
         }
-        mReader.ExitContainer(mNestingEnters[--mCurrentNesting]);
+        CHIP_ERROR err = mReader.ExitContainer(mNestingEnters[--mCurrentNesting]);
+        if (err != CHIP_NO_ERROR)
+        {
+            mValueBuilder.AddFormat("ERROR: %" CHIP_ERROR_FORMAT, err.Format());
+            mNameBuilder.AddFormat("EXIT CONTAINER");
```
Comment (tcarmelveilleux) [inline src/lib/format/protocol_decoder.cpp:299]: END Container

**Review by tcarmelveilleux — APPROVED** [review]

File: src/tracing/json/json_tracing.cpp (hunk 6 above)
Comment (andy31415) [reply src/tracing/json/json_tracing.cpp]: Updated to `Anonymous<>` (and injected index in there)

**Pull request merged**